elixir/apps/domain/lib/domain/config/clients_upstream_dns.ex

```diff
@@ -0,0 +1,105 @@
+defmodule Domain.Config.ClientsUpstreamDNS do
+  @moduledoc """
+  Embedded Schema for Clients Upstream DNS
+  """
+  use Domain, :schema
+  import Domain.Validator
+  import Ecto.Changeset
+
+  @dns_types ~w[ip dns_over_tls dns_over_http]
+
+  @primary_key false
+  embedded_schema do
+    field :type, :string
```

Comment:
Agreed with @conectado to call it a `:protocol`.

Reply:
I've updated everything to use `:protocol`, rather than `:type` as recommended.